rosidl_generator_cpp/resource/msg__traits.hpp.template

```diff
@@ -0,0 +1,70 @@
+// generated from rosidl_generator_cpp/resource/msg__traits.hpp.template
+
+@#######################################################################
+@# EmPy template for generating <msg>__traits.hpp files
+@#
+@# Context:
+@#  - spec (rosidl_parser.MessageSpecification)
+@#    Parsed specification of the .msg file
+@#  - subfolder (string)
+@#    The subfolder / subnamespace of the message
+@#    Either 'msg' or 'srv'
+@#  - get_header_filename_from_msg_name (function)
+@#######################################################################
+@
+@{
+from rosidl_generator_cpp import MSG_TYPE_TO_CPP
+
+cpp_namespace = '%s::%s::' % (spec.base_type.pkg_name, subfolder)
+}@
+#include <array>
+#include <stdint.h>
+#include <string>
+#include <vector>
+
+namespace rosidl_generator_traits
+{
+
+#ifndef __ROSIDL_GENERATOR_CPP_TRAITS
+#define __ROSIDL_GENERATOR_CPP_TRAITS
+
+template<typename T>
+struct has_fixed_size {};
+
+#endif  /* __ROSIDL_GENERATOR_CPP_TRAITS */
+
+#ifndef __@(spec.base_type.pkg_name)__@(subfolder)__@(get_header_filename_from_msg_name(spec.base_type.type))__traits__hpp__
+#define __@(spec.base_type.pkg_name)__@(subfolder)__@(get_header_filename_from_msg_name(spec.base_type.type))__traits__hpp__
+
+#include "@(spec.base_type.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.base_type.type))__struct.hpp"
+
+@{
+fixed_template_strings = []
+fixed = True
+fixed_template_string = ''
```

Comment:
This definition is not necessary since the variable gets set later in all cases.